src/features/auth/sign-in/model/use-sign-in.ts

```diff
@@ -0,0 +1,29 @@
+'use client';
+
+import { useState } from 'react';
+import { FirebaseError } from 'firebase/app';
+import { apiSignIn } from '@/shared/api/firebase/auth';
+
+type Payload = { email: string; password: string };
+
+export function useSignIn() {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  async function mutate({ email, password }: Payload) {
+    setLoading(true);
+    setError(null);
+    try {
+      const cred = await apiSignIn({ email, password });
+      return cred.user;
+    } catch (e) {
+      const code = e instanceof FirebaseError ? e.code : undefined;
+      setError(code ?? 'unknown');
```

Comment:
I'm not sure, but it seems to me that the version I proposed has the same functionality, while saving us from ??

```suggestion
      const code = e instanceof FirebaseError ? e.code : 'unknown';
      setError(code);
```

Reply:
This comment makes sense if you decide to keep this file after all.